src/app/vm/components/vm-snapshots-sidebar-card/vm-snapshots-sidebar-card.component.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { EntityAction } from '../../../shared/interfaces';
+import { VmSnapshot } from '../../../shared/models/vm-snapshot.model';
+
+@Component({
+  selector: 'cs-vm-snapshots-card',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './vm-snapshots-sidebar-card.component.html',
+  styleUrls: ['./vm-snapshots-sidebar-card.component.scss'],
+})
+export class VmSnapshotsSidebarCardComponent implements OnInit {
+  @Input()
+  public lastVmSnapshot: VmSnapshot;
+  @Input()
+  public vmSnapshotsCount: number;
```

Comment:
Can you just pass all snapshots? You can then get the number of snapshots and the last snapshot.

Reply:
This is because I decided to extract logic from the component and make it dumb.